src/Language/PureScript/Docs/Convert/Single.hs

```diff
@@ -106,6 +108,26 @@ augmentDeclarations (partitionEithers -> (augments, toplevels)) =
       mergeComments kc Nothing = kc
       mergeComments (Just kcoms) (Just dcoms) =
         Just $ kcoms <> "\n" <> dcoms
+  augmentWith (AugmentRole comms roles) d =
+    d { declComments = mergeComments (declComments d) comms
+      , declInfo = insertRoles
+      }
+    where
+      mergeComments Nothing rc = rc
+      mergeComments dc Nothing = dc
+      mergeComments (Just dcoms) (Just rcoms) =
+        Just $ dcoms <> "\n" <> rcoms
```

Comment:
I've opted to merge the role annotation's comments under the declaration's comments. Depending on how we visualize content, we might want to store the role annotation comments in the `declInfo` field, too.